amoro-format-iceberg/src/main/java/org/apache/amoro/formats/mixed/MixedCatalog.java

```diff
@@ -76,9 +77,13 @@ public AmoroTable<?> loadTable(String database, String table) {
 
   @Override
   public List<String> listTables(String database) {
-    return catalog.listTables(database).stream()
-        .map(TableIdentifier::getTableName)
-        .collect(Collectors.toList());
+    try {
+      return catalog.listTables(database).stream()
+          .map(TableIdentifier::getTableName)
+          .collect(Collectors.toList());
+    } catch (NoSuchDatabaseException e) {
+      throw new NoSuchDatabaseException(e, database);
```

Comment:
Why do we catch and retry the original exception here

Reply:
The 'listTables' called in BasicMixedCatalog、BasicMixedIcebergCatalog、InternalMixedIcebergCatalog , etc., will throw other exceptions  such ad IllegalStateException not NoSuchDatabaseException.

Reply:
> Why do we catch and retry the original exception here

Thanks for your question. The 'listTables' called in BasicMixedCatalog、BasicMixedIcebergCatalog、InternalMixedIcebergCatalog , etc., will throw other exceptions  such as IllegalStateException not NoSuchDatabaseException.

Reply:
> Why do we catch and retry the original exception here

We can not catch this exception here. I have changed it.